### DISCUSSION TITLE
Alpine-related DOM diffing issue when navigating back

### DISCUSSION THREAD

**Discussion by stancl** [livewire/livewire#7947] [7. Bugs (archived)] [answered]
### Livewire version

v3.4.4

### Laravel version

v10.3.3

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Use `wire:navigate`
2. Create an Alpine component that sets classes dynamically based on some variable
3. Navigate forward a few times (1 navigation is enough, and you can navigate *forward* as much as you want)
4. Navigate back just once
5. The component becomes broken: classes that were only added by Alpine are now "hardcoded" into the template.

It's possible that this requires a specific Alpine component/template for the bug to happen.

This has only started happening after upgrading from LW 2 to LW 3. In both setups Alpine v3 was used, so the issue is somewhere in how Livewire handles these cases when using `wire:navigate`.

### Repository or Wirebox URL

https://github.com/stancl/livewire-bug-repro1

### Additional information

Here's how it works. Notice I can navigate forward infinitely and the component never breaks. Once I navigate back, the dropdown breaks.

https://github.com/livewire/livewire/assets/33033094/d208790e-fff3-4e1f-93ab-e04e01d828a0

The way it breaks is that the `hidden` becomes "hardcoded", and then `block` is dynamically added based on state:

https://github.com/stancl/livewire-bug-repro1/blob/be52894b15942e9cafc49621a12974e1e58e98fe/resources/views/components/header.blade.php#L31

https://github.com/livewire/livewire/assets/33033094/aff6d201-7fc5-4b01-9496-f039a311642a

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by danie-ramdhani** [comment]
mmm... yeah.... it's bug for been ages (i forget the discussion link) and has workaround using alpine class object:

```blade
...
<div class="lg:hidden" id="mobile-menu" x-cloak
         :class="{
             'block': mobileMenuOpen,
             'hidden': !mobileMenuOpen
         }">
...
```

**Reply by danie-ramdhani** [reply, marked as answer]
Made a PR https://github.com/livewire/livewire/pull/8154

**Reply by danie-ramdhani** [reply]
PR is merged with fix bug, should be fixed in next release.
